tutorials/metrics.md

```diff
@@ -39,22 +39,34 @@ metrics_key = "/path/to/key.pem"
+| `crio_image_layer_reuse`                         | `name`                                                                                                                                  | Counter   | (DEPRECATED) Reused (not pulled) local image layer count by name.                                                 |
+| `crio_containers_oom_total`                      |                                                                                                                                         | Counter   | Total number of containers killed because they ran out of memory (OOM)                                            |
+| `crio_containers_oom`                            | `name`                                                                                                                                  | Counter   | (DEPRECATED) Containers killed because they ran out of memory (OOM) by their name                                 |
+| `crio_processes_defunct`                         |                                                                                                                                         | Gauge     | Total number of defunct processes in the node                                                                     |
+| `crio_operations_total`                          | every CRI-O RPC\*                                                                                                                       | Counter   | Cumulative number of CRI-O operations by operation type.                                                          |
+| `crio_operations_latency_seconds_total`          | every CRI-O RPC\*,<br><br>`network_setup_pod` (CNI pod network setup time),<br><br>`network_setup_overall` (Overall network setup time) | Summary   | Latency in seconds of CRI-O operations. Split-up by operation type.                                               |
+| `crio_operations_latency_seconds`                | every CRI-O RPC\*                                                                                                                       | Gauge     | Latency in microseconds of individual CRI calls for CRI-O operations. Broken down by operation type.              |
+| `crio_operations_errors_total`                   | every CRI-O RPC\*                                                                                                                       | Counter   | Cumulative number of CRI-O operation errors by operation type.                                                    |
+| `crio_image_pulls_by_digest_bytes_total`         | `name`, `digest`, `mediatype`, `size`                                                                                                   | Counter   | Bytes transferred by CRI-O image pulls by digest.                                                                 |
```

Comment:
I'm a bit worried about the cardinality of these. Especially digest seems like something that could change very frequently and each dimension will stay exposed indefinitely.

Reply:
Oh and size of course. Is that bounded? Otherwise this can't really be reasonably exposed: https://www.robustperception.io/cardinality-is-key

Reply:
it's likely digest will have quite high cardinality... For size we could sort into buckets instead of reporting the size directly

Reply:
@haircommander @saschagrunert 
Sorting sizes into buckets seems like a reasonable way to reduce cardinality of size. What bucket sizes would be preferred? (any reference implementations?)
Also, we can decide if we really wanna drop digest altogether.

Reply:
There is `crio_image_pulls_layer_size` already, which is essentially a counter of pulls by size.
Digest probably needs to get dropped

Reply:
Label `digest` was dropped as also stated beneath.
> Introduced metric names as
> `crio_image_pulls_bytes_total` and `crio_image_pulls_skipped_bytes_total` that follow naming best practices. Dropped high cardinality labels like `name`, `digest`. However, `size` is being used but with sorted buckets so as to keep cardinality in check.